File: Coursework/Data Structures and Algorithms/Implementations/Algorithms/Selection Sort.cpp
#include <bits/stdc++.h>

using namespace std;

/// Selection Sort O(N^2)

int main() {
    int x[] = {68, 894, 15, -20 , 8};
    int n = sizeof(x)/sizeof(x[0]);

    for(int i=0;i<n;i++){
        int mn=INT_MAX, mni;

        for(int j=i;j<n;j++){
            if(x[j]<mn){
                mn=x[j];
                mni=j;
            }
        }

        swap(x[mni],x[i]);
    }

    for(auto u:x)cout<<u<<" ";
}

File: Data Structures and Algorithms/Implementations/Algorithms/Selection Sort.java
// Selection Sort O(n^2), In-Place, not stable.

public class Main {

    private static void swap(int[] x, int a, int b) {
        int t = x[a];
        x[a] = x[b];
        x[b] = t;
    }

    public static void main(String[] args) {
        int[] x = {5, 9, 1, 0, -561, 69, 98, -20, 6};
        for (int i = 0; i < x.length; i++) {
            int mn = x[i], mni = i; // Min value, Min value index.
            for (int j = i + 1; j < x.length; j++) {
                if (x[j] < mn) {
                    mn = x[j];
                    mni = j;
                }
            }
            swap(x, mni, i);
        }
        for (int i : x) System.out.print(i + " ");
    }
}